File: problem69.py
class Solution:
    def mySqrt(self, x: int):
        # Base CAse
        if x == 0:
            return 0
        if x < 4:
            return 1
        # Narrow down into number
        for i in range(1, x//2+2):
            if i * i == x:  
                return i
            elif i*i > x:
                return i-1

File: problem69.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int mySqrt(int x) {
        // Binary search
        // Base Case

        if(x < 2){
            return x;
        } 

        long left = 1;
        long right = x/2;

        while( left <= right){
            long mid = (left + right) / 2;
            long sqr = mid * mid;

            if(sqr == x){
                return mid;
            }
            else if(sqr > x){
                right = mid -1;
            }
            else{
                left = mid + 1;
            }
        }
        return right;
    }    
};